pkg/broadcaster/broadcaster.go

```diff
@@ -0,0 +1,158 @@
+package broadcaster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+	"time"
+)
+
+type client struct {
+	w    io.WriteCloser
+	msgs chan string
+	done chan struct{}
+}
+
+func (c *client) Close() error {
+	// TODO: Consider making the timeout configurable.
+	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond) //nolint:mnd // Arbitrary duration.
+	defer cancel()
+
+	close(c.msgs)
+	select {
+	case <-ctx.Done():
+		_ = c.w.Close() // Best effort.
+		return context.DeadlineExceeded
```

Comment:
Let me see if I got it right:

1. Closing the BufferedBroadcaster closes the clients
2. Closing a client closes the msgs channel
3. A closed msgs channel should make the Write goroutine exit
4. That was to happen in 500ms, otherwise the client's io.WriterCloser is closed, preventing further writes and making the Write goroutine abort

This seems like an easy avenue for a slow client to lose data once a job stops, as we are somewhat negating the advantages of the buffer in the msgs channel.

I think a simple solution is to simply not take ownership of the client's io.WriterClose - ie, take only an io.Writer when subscribing and let the client handle the closing.